src/app/Interface/singleproduct.ts

```diff
@@ -0,0 +1,29 @@
+export interface IOneSingleProduct {
```

Comment:
Would it be okay for you to rename this to IProduct?
The fact that its a only one product is stated by "Product" which is singular. If the method would return more then one product the name should contain the plural "Products".

Reply:
the name "IProduct" is already taken, by another interface, the Interface was renamed to "IDetailedProduct" and now it extends the "IProduct" Interface.